server/src/Persistence.hs

```diff
@@ -88,9 +145,61 @@ load config = runExceptT $ do
   recoverJournal val
   return val
 
+syncToBackend :: StorageBackend -> PersistentValue -> IO ()
+syncToBackend File pv = syncFile pv
+syncToBackend Sqlite pv = syncSqliteFile pv
+
+-- * SQLite loading and syncing
+
+-- | There is currently just one row and it contains only one column of type SBS.ByteString.
+-- This single field holds the whole JSON value for now.
+data JsonRow = JsonRow {jsonByteString :: SBS.ByteString} deriving (Show)
+
+instance FromRow JsonRow where
+  fromRow = JsonRow <$> field
+
+-- | Read and decode the Sqlite data file from disk
+readSqliteData :: FilePath -> ExceptT String IO Store.Value
+readSqliteData filePath = ExceptT $ do
+  -- read the data from SQLite
+  conn <- liftIO $ open filePath
+  jsonRows <- liftIO $ (query_ conn "SELECT * from icepeak" :: IO [JsonRow])
+
+  case jsonRows of
+    -- if there is no data yet, we simply return the empty object. We do the same thing for the
+    -- file backend.
```

Comment:
This is no longer true, the empty object is now created elsewhere and read assumes it exists, right?

Reply:
Correct, I updated the comment.